gen_cert.py

```diff
@@ -80,36 +81,23 @@
 }
 
 
-def prettify_isodate(isoformat_date):
-    """Convert a string like '2012-02-02' to one like 'February 2nd, 2012'"""
-    m = RE_ISODATES.match(isoformat_date)
-    if not m:
-        raise TypeError("prettify_isodate called with incorrect date format: %s" % isoformat_date)
-    day_suffixes = {'1': 'st', '2': 'nd', '3': 'rd', '21': 'st', '22': 'nd', '23': 'rd', '31': 'st'}
-    months = ['January', 'February', 'March', 'April', 'May', 'June', 'July',
-              'August', 'September', 'October', 'November', 'December']
-    date = {'year': '', 'month': '', 'day': '', 'suffix': 'th'}
-    date['year'] = m.group('year')
-    date['month'] = months[int(m.group('month')) - 1]
-    date['day'] = m.group('day').lstrip('0')
-    date['suffix'] = day_suffixes.get(date['day'], 'th')
-    return "%(month)s %(day)s%(suffix)s, %(year)s" % date
-
-
-def get_cert_date(calling_date_parameter, configured_date_parameter):
+def get_cert_date(calling_date_parameter, configured_date_parameter, date_format='%B %d, %Y', locale="en_US"):
     """Get pertinent date for display on cert
 
     - If cert passes a set date in 'calling_date_parameter', format that
     - If using the "ROLLING" certs feature, use today's date
     - If all else fails use 'configured_date_parameter' for date
     """
+
+    setlocale(LC_TIME, locale)
```

Comment:
## 22.2.1. Background, details, hints, tips and caveats [1]

> It is generally a bad idea to call setlocale() in some library routine, since as a side effect it affects the entire program. Saving and restoring it is almost as bad: it is expensive and affects other threads that happen to run before the settings have been restored.

Python's standard datetime handling has its share of drawbacks. We may want to use an external library here; `Babel` looks promising [2].

```python
from babel.dates import format_date
date_value = format_date(calling_date_parameter, format='long', locale=locale)
# u'June 25, 2015'
# u'25 de junio de 2015'
# u'25 juin 2015'
```



[1] https://docs.python.org/2/library/locale.html#background-details-hints-tips-and-caveats
[2] http://babel.pocoo.org/docs/dates/